### PR TITLE
[GraphBolt] de-duplicate code for reading data from achive

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: graphbolt/include/graphbolt/serialize.h [hunk 1]
```diff
@@ -66,6 +66,40 @@ namespace graphbolt {
...
     torch::serialize::InputArchive& archive, const std::string& key);
 
+/**
+ * @brief Read dictionary from archive.
+ * @param archive Input archive.
+ * @param key Key name of dictionary.
+ *
+ * @return dictionary.
+ */
+template <typename DictT>
```
Comment (czkkkkkk) [inline graphbolt/include/graphbolt/serialize.h:76]: The PR makes a lot of sense to me. A minor suggestion is to make both key and value types as the templates for this function.
```cpp
template <typename KeyType, typename ValueType>
```
Then the function is general to all dictionary types and we don't need to specially handle `int64_t` and `torch::Tensor`.

**Review by czkkkkkk — APPROVED** [review]

File: graphbolt/include/graphbolt/serialize.h (hunk 1 above)
Comment (Rhett-Ying) [reply graphbolt/include/graphbolt/serialize.h:76]: I wish I could achieve. The major blocker is the caller function name when converting IValue to real data such as `toInt()`, `toTensor()`. Is there any template-like calls for these ones? For example, `to<int>`, `to<Tensor>`?

Comment (czkkkkkk) [reply graphbolt/include/graphbolt/serialize.h:76]: I see. I am not sure whether [this](https://github.com/pytorch/pytorch/blob/4ea7430ffb0b37e84108c70f90f402f5fd37262e/aten/src/ATen/core/ivalue.h#L1004) is workable.

Comment (Rhett-Ying) [reply graphbolt/include/graphbolt/serialize.h:76]: this works. let me further refine related code.

Comment (Rhett-Ying) [reply graphbolt/include/graphbolt/serialize.h]: I used generalized `to<>` for further de-duplication

**Pull request merged**